packages/data-service/src/data-service.ts

```diff
@@ -666,7 +666,95 @@ export interface DataService {
   configuredKMSProviders(): string[];
 }
 
-export class DataServiceImpl implements DataService {
+const maybePickNs = ([ns]: unknown[]) => {
+  if (typeof ns === 'string') {
+    return { ns };
+  }
+};
+
+const isPromise = <T>(val: any): val is Promise<T> => {
```

Comment:
```suggestion
const isThenable = <T>(val: any): val is Thenable<T> => {
```

(because that’s what it checks…? :))

Reply:
Yes, thanks! (I updated your comment so that I can accept right from here)